### PR TITLE
chore: fix retry docs and add example

### PR DESCRIPTION
**Pull request by elsirion**

### CODE REVIEW

File: fedimint-core/src/util/mod.rs [hunk 1]
```diff
@@ -323,14 +344,14 @@ pub use backon::{
...
-    statergy: impl backon::BackoffBuilder,
+    strategy: impl backon::BackoffBuilder,
     op_fn: F,
 ) -> Result<T, anyhow::Error>
 where
     F: Fn() -> Fut,
     Fut: Future<Output = Result<T, anyhow::Error>>,
 {
-    let mut statergy = statergy.build();
+    let mut statergy = strategy.build();
```
Comment (bradleystachurski) [inline fedimint-core/src/util/mod.rs:354]: nit: s/statergy/strategy

File: fedimint-core/src/util/mod.rs [hunk 2]
```diff
@@ -313,8 +313,29 @@ pub use backon::{
...
-/// Run the supplied closure `op_fn` up to `max_attempts` times. Wait for the
-/// supplied `Duration` `interval` between attempts
+/// Run the supplied closure `op_fn` until it succeeds. Frequency and number of
+/// retries is determined by the specified strategy.
+///
+/// ```
+/// use std::time::Duration;
+///
+/// use fedimint_core::util::{retry, FibonacciBackoff};
+/// # tokio_test::block_on(async {
```
Comment (bradleystachurski) [inline fedimint-core/src/util/mod.rs:323]: Looks like udeps has issues with doc-tests so we'll need to update `fedimint-core/Cargo.toml`

```
[package.metadata.cargo-udeps.ignore]
development = ["tokio-test"]
```

https://github.com/est31/cargo-udeps?tab=readme-ov-file#ignoring-some-of-the-dependencies

**Review by bradleystachurski — DISMISSED** [review]

File: fedimint-core/src/util/mod.rs (hunk 2 above)
Comment (elsirion) [reply fedimint-core/src/util/mod.rs:323]: Thx for the pointer, saved me a lot of time!

File: fedimint-core/Cargo.toml [hunk 3]
```diff
@@ -67,12 +67,16 @@ async-lock = "3.3"
...
 [dev-dependencies]
-test-log = { version = "0.2", features = [ "trace" ], default-features = false }
+test-log = { version = "0.2", features = ["trace"], default-features = false }
 once_cell = "1.19.0"
-tracing-subscriber = { version = "0.3.18", features = [ "env-filter" ] }
+tracing-subscriber = { version = "0.3.18", features = ["env-filter"] }
+tokio-test = "0.4.4"
+
+[package.metadata.cargo-udeps.ignore]
+development = ["tokio-test"]
```
Comment (maan2003) [inline fedimint-core/Cargo.toml:82]: missing new line

**Review by maan2003 — DISMISSED** [review]

File: fedimint-core/Cargo.toml (hunk 3 above)
Comment (elsirion) [reply fedimint-core/Cargo.toml]: fixed

**Review by maan2003 — APPROVED** [review]

File: fedimint-core/Cargo.toml [hunk 4]
```diff
@@ -26,30 +26,30 @@ bls12_381 = "0.7.1"
 serdect = "0.2.0"
 itertools = { workspace = true }
 jsonrpsee-types = { version = "0.22.2" }
-jsonrpsee-core = { version = "0.22.2", features = [ "client" ] }
+jsonrpsee-core = { version = "0.22.2", features = ["client"] }
```
Comment (maan2003) [inline fedimint-core/Cargo.toml:29]: unrelated: which one do you like more?
I prefer the top one in toml because it "feels right"

Comment (elsirion) [reply fedimint-core/Cargo.toml:29]: I don't care, wasn't this done by our `just format`? Is my IDE suddenly formatting toml files?! If so I'm sorry, I'll teach it some manners.

Comment (maan2003) [reply fedimint-core/Cargo.toml:29]: yep, it is your IDE. The changes are good, we need toml formatter in fedimint

**Pull request merged**